src/auth/integration-tests/src/lib.rs

```diff
@@ -183,6 +177,60 @@ pub async fn workload_identity_provider_url_sourced() -> anyhow::Result<()> {
     Ok(())
 }
 
+pub async fn workload_identity_provider_executable_sourced() -> anyhow::Result<()> {
+    let project = std::env::var("GOOGLE_CLOUD_PROJECT").expect("GOOGLE_CLOUD_PROJECT not set");
+    let audience = get_oidc_audience();
+    let (service_account, client_email) = get_byoid_service_account_and_email();
+
+    let id_token = generate_id_token(audience.clone(), client_email, service_account).await?;
+
+    let source_token_output_file = serde_json::json!({
+        "success": true,
+        "version": 1,
+        "token_type": "urn:ietf:params:oauth:token-type:jwt",
+        "id_token": id_token,
+    })
+    .to_string();
+
+    let file = tempfile::NamedTempFile::new().unwrap();
+    let path = file.into_temp_path();
+    std::fs::write(&path, source_token_output_file)
+        .expect("Unable to write to temp file with id token");
+
+    let path = path.to_str().unwrap();
+    let contents = serde_json::json!({
+      "type": "external_account",
+      "audience": audience,
+      "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+      "token_url": "https://sts.googleapis.com/v1/token",
+      "credential_source": {
+        "executable": {
+            "command": format!("cat {path}"),
+            "output_file": path,
+        },
+      }
```

Comment:
I think you want the program to generate the file, otherwise I am not sure the test matches its title?

Reply:
make sense, just pushed an update setting just the `command` and enabling execution with env var.